event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinNviReportEventConsumer.java

```diff
@@ -109,6 +111,20 @@ private DbCandidate createUpdatedDbCandidate(
     return historicalCandidate.copy().publicationDetails(updatedPublicationDetails).build();
   }
 
+  private List<DbOrganization> getCurrentTopLevelOrganizations(PublicationDto publication) {
+    if (isMissing(publication.topLevelOrganizations())) {
+      // All NVI results should have topLevelOrganizations affiliated, but some
+      // imported results are missing data.
+      logger.error(
+          "Missing top level organizations for publication with identifier {}",
+          publication.identifier());
+      return emptyList();
+    }
+    return publication.topLevelOrganizations().stream()
+        .map(Organization::toDbOrganization)
+        .toList();
+  }
```

Comment:
Typically, I'd recommend not commenting, but use the method name. Also, single return points are nice.

```suggestion
    return isMissing(publication.topLevelOrganizations()) 
      ? createEmptyListForImportedResultsThatLackData() 
      : publication.topLevelOrganizations().stream()
        .map(Organization::toDbOrganization)
        .toList();
  }

  private List<DbOrganization> createEmptyListForImportedResultsThatLackData() {
      logger.error(
          "Missing top level organizations for publication with identifier {}",
          publication.identifier());
      return emptyList();
    }
```